pkg/validate/validate_test.go

```diff
@@ -52,31 +52,31 @@ func TestClusterName(t *testing.T) {
 func TestSubnetCIDR(t *testing.T) {
 	cases := []struct {
 		cidr  string
-		valid bool
+		valid string
 	}{
-		{"0.0.0.0/32", false},
-		{"1.2.3.4/0", false},
-		{"1.2.3.4/1", false},
-		{"1.2.3.4/31", true},
-		{"1.2.3.4/32", true},
-		{"0:0:0:0:0:1:102:304/116", false},
-		{"0:0:0:0:0:ffff:102:304/116", true},
-		{"172.17.1.2/20", false},
-		{"172.17.1.2/8", false},
-		{"255.255.255.255/1", false},
-		{"255.255.255.255/32", true},
+		{"0.0.0.0/32", "address must be specified"},
+		{"1.2.3.4/0", "invalid network address. got 1.2.3.4/0, expecting 0.0.0.0/0"},
+		{"1.2.3.4/1", "invalid network address. got 1.2.3.4/1, expecting 0.0.0.0/1"},
+		{"1.2.3.4/31", "blah"},
+		{"1.2.3.4/32", "blah"},
+		{"0:0:0:0:0:1:102:304/116", "must use IPv4"},
+		{"0:0:0:0:0:ffff:102:304/116", "invalid network address. got 1.2.3.4/20, expecting 1.2.0.0/20"},
+		{"172.17.0.0/20", "overlaps with default Docker Bridge subnet (172.17.0.0/20)"},
+		{"172.0.0.0/8", "overlaps with default Docker Bridge subnet (172.0.0.0/8)"},
+		{"255.255.255.255/1", "invalid network address. got 255.255.255.255/1, expecting 128.0.0.0/1"},
+		{"255.255.255.255/32", "blah"},
 	}
 	for _, tc := range cases {
 		t.Run(tc.cidr, func(t *testing.T) {
-			_, cidr, err := net.ParseCIDR(tc.cidr)
+			ip, cidr, err := net.ParseCIDR(tc.cidr)
 			if err != nil {
 				t.Fatalf("could not parse cidr: %v", err)
 			}
-			err = SubnetCIDR(cidr)
-			if tc.valid {
-				assert.NoError(t, err)
+			err = SubnetCIDR(&net.IPNet{IP: ip, Mask: cidr.Mask})
+			if err != nil {
```

Comment:
This is [still][1] switching on the returned error and not on a testcase property.  That can cause false negatives, e.g. if `ParseCIDR` fails to return an expected error, then we'll hit the `err == nil` case and execute the tautological `assert.NoError(t, err)`.  I've pushed `git://github.com/wking/openshift-installer.git subnet-cidr-test-fix` (4c9deaebe4) with a fixup to switch on the expected error instead.  Can you take a look and squash that into your commit if it looks good to you (or tell me why it doesn't look good ;)?

[1]: https://github.com/openshift/installer/pull/1065#discussion_r247736725

Reply:
done with https://github.com/openshift/installer/commit/df680a1ef30ace52fc9f72966335b0228c342856 -> https://github.com/openshift/installer/commit/5c29e90536dec16718461d04674f751f6d1a0456